osu.Game/Beatmaps/Formats/OsuLegacyDecoder.cs

```diff
@@ -331,6 +332,7 @@ private void handleTimingPoints(Beatmap beatmap, string line)
 
             if (speedMultiplier != difficultyPoint.SpeedMultiplier)
             {
+                beatmap.ControlPointInfo.DifficultyPoints.RemoveAll(x => x.Time == time);

```

Comment:
Aha I see, good catch. Since stable doesn't strip them at the moment either, let's keep everything else as it is.